### PR TITLE
Add generation of static HTML docs served from API server

### PR DESCRIPTION
**Pull request by jimmidyson**
When run with `java -jar target/swarm-runtime.jar`, static HTML docs are served
from http://localhost:8080. Haven't figured out how to get those embedded in dev
mode, but that's less important anyway.

### CODE REVIEW

**Comment by jimmidyson** [comment]
@kahboom Hope that this should produce some static HTML docs for the REST API to make it easier for you to review them as needed & mean we don't need to manually update them as the implementation progresses. @KurtStam it does mean we need to keep the swagger annotations up to date & verbose though :)

**Comment by jimmidyson** [comment]
Had to use a SNAPSHOT dependency plugin to get circle to pass (old version is not Maven 3 compatible it seems). New version should be released soon.

**Comment by jimmidyson** [comment]
See example static doc output at https://32-75404146-gh.circle-artifacts.com/0/tmp/circle-artifacts.W198HOB/apidocs.html. We can customize templates as we need but we can sort that out later,

File: runtime/src/main/java/com/redhat/ipaas/rest/Connections.java [hunk 1]
```diff
@@ -30,58 +33,61 @@
...
-			@ApiParam(value = "id of the connection", required = true) @PathParam("id") String id) {
-		dataMgr.delete(Connection.class, id);
-		
-	}
-	
-	
-	
-}
\ No newline at end of file
+    @Inject
```
Comment (rhuss) [inline runtime/src/main/java/com/redhat/ipaas/rest/Connections.java]: We should settle on a convention of the indent in Java source files. I'd prefer 4 spaces overall, (not 2 for the initial one) but would anything we agree upon. But we should stick to the same convention to not mess up diffs.

Comment (jimmidyson) [inline runtime/src/main/java/com/redhat/ipaas/rest/Connections.java]: Agreed 4 spaces for Java.

Sorry about the formatting changes - got any tips on something to check style (checkstyle I assume?) that works well & is not too intrusive? P.S. I love gofmt :)

Comment (jimmidyson) [inline runtime/src/main/java/com/redhat/ipaas/rest/Connections.java]: I've set that up in `.editorconfig` already (https://github.com/redhat-ipaas/ipaas-api-java/blob/master/.editorconfig#L8-L9). See http://editorconfig.org if you don't know about editorconfig (it's got pretty wide editor support via plugins & is nice & simple).

Comment (jimmidyson) [inline runtime/src/main/java/com/redhat/ipaas/rest/Connections.java]: Yeah I use intellij too. Raised #20 to discuss more.

File: pom.xml [hunk 2]
```diff
@@ -100,6 +122,7 @@
     <maven.resources.plugin.version>3.0.1</maven.resources.plugin.version>
     <maven.source.plugin.version>2.4</maven.source.plugin.version>
     <maven.surefire.plugin.version>2.19.1</maven.surefire.plugin.version>
+    <maven.dependency.plugin.version>3.0.0-SNAPSHOT</maven.dependency.plugin.version>
```
Comment (iocanel) [inline pom.xml]: Do we even use the maven dependency plugin?
And why do we need a snapshot version.

I am sure there is a reason, just being curious.

Comment (jimmidyson) [inline pom.xml]: For circleci to speed up caching dependencies between builds `mvn dependency:go-offline` happens in dependency stage before cache is created for next build, but before actual build is run (don't want to cache build artifacts from this build).

**Comment by jimmidyson** [comment]
Anything need fixing up? Please can someone approve the PR & merge it if you are or tell me what needs changing if not. Pretty please?

**Comment by KurtStam** [comment]
Sure - taking a look now.

**Comment by KurtStam** [comment]
[merge]

**Comment by jimmidyson** [comment]
Sorry no CI for this yet on Jenkins - I can set that up tonight.

**Pull request merged**